src/nvim/edit.h

```diff
@@ -12,6 +13,14 @@
 #define CPT_INFO        3   // "info"
 #define CPT_COUNT       4   // Number of entries
 
+// lua functions used to filter completion matches
+// global_filterfunc is used for all completion sources,
+// while user_filterfunc is only used to filter matches
+// from vim.api.nvim_complete().
+EXTERN LuaRef global_filterfunc INIT(= LUA_NOREF);
+EXTERN LuaRef user_filterfunc INIT(= LUA_NOREF);
+EXTERN LuaRef active_filterfunc INIT(= LUA_NOREF);
```

Comment:
would be nice if these are grouped a `compl_filter` struct. Then all references look like `compl_filter.user_func`, etc.